src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,73 @@
-import React from 'react';
+import React, { ChangeEvent, FormEvent, useState } from 'react';
+import classNames from 'classnames';
+
+import { Movie } from '../../types/Movie';
+import { getMovie } from '../../api';
+import { MovieCard } from '../MovieCard';
+
 import './FindMovie.scss';
 
-export const FindMovie: React.FC = () => {
+interface Props {
+  onAddMovie: (movie: Movie) => void;
+}
+
+export const FindMovie: React.FC<Props> = ({ onAddMovie }) => {
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [query, setQuery] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setLoading(true);
+
+    getMovie(query.trim())
+      .then(response => {
+        if ('Error' in response) {
+          setError(true);
+
+          return;
+        }
+
+        const poster = response.Poster === 'N/A'
+          ? 'https://via.placeholder.com/360x270.png?text=no%20preview'
+          : response.Poster;
+
+        const foundedMovie = {
+          title: response.Title,
+          description: response.Plot,
+          imgUrl: poster,
+          imdbUrl: `https://www.imdb.com/title/${response.imdbID}`,
+          imdbId: response.imdbID,
+        };
+
+        setMovie(foundedMovie);
+      })
+      .finally(() => setLoading(false));
+  };
+
+  const handleAddMovie = () => {
+    if (!movie) {
```

Comment:
you already handle it here
![image](https://github.com/mate-academy/react_movies-list-fetch-movies/assets/91411692/57ea832a-517b-4691-8183-30de85866664)

```suggestion
```